examples/lang/erc20/src/lib.rs

```diff
@@ -135,16 +135,16 @@ contract! {
 
     impl Erc20 {
         /// Decreases the allowance and returns if it was successful.
-        fn try_decrease_allowance(&mut self, from: &AccountId, to: &AccountId, by: Balance) -> bool {
+        fn try_decrease_allowance(&mut self, from: &AccountId, by: Balance) -> bool {
             // The owner of the coins doesn't need an allowance.
             if &env::caller() == from {
                 return true
             }
-            let allowance = self.allowance_or_zero(from, to);
+            let allowance = self.allowance_or_zero(from, &env::caller());
             if allowance < by {
                 return false
             }
-            self.allowances.insert((*from, *to), allowance - by);
+            self.allowances.insert((*from, env::caller()), allowance - by);
```

Comment:
unless this is already handled by the `Sub` implementation for balance, shouldn't this be a checked sub?

Reply:
This configuration line:
https://github.com/paritytech/ink/blob/master/examples/lang/erc20/.cargo/config#L3

ensures that all math in the contract is "safe math" i.e. any overflow/underflow will revert the transaction.

Something we will need to teach to upcoming ink! developers.

Please look for other issues though!

Reply:
ah, nice one. :open_hands: